packages/plugin-svelte/src/index.ts

```diff
@@ -0,0 +1,74 @@
+import path from 'path';
+import { logger } from '@rsbuild/shared';
+import type { RsbuildPlugin, RsbuildPluginAPI } from '@rsbuild/core';
+
+export type PluginSvelteOptions = {};
+
+export function pluginSvelte(
+  options: PluginSvelteOptions = {},
+): RsbuildPlugin<RsbuildPluginAPI> {
+  return {
+    name: 'plugin-svelte',
+
+    async setup(api) {
+      let sveltePath: string = '';
+      try {
+        // Resolve `svelte` package path from the project directory
+        sveltePath = path.dirname(
+          require.resolve('svelte/package.json', {
+            paths: [api.context.rootPath],
+          }),
+        );
+      } catch (err) {
+        logger.error(
+          `Cannot resolve \`svelte\` package under the project directory, did you forget to install it?`,
+        );
+        throw new Error(`Cannot resolve \`svelte\` package`, {
+          cause: err,
+        });
+      }
+
+      api.modifyRsbuildConfig((config, { mergeRsbuildConfig }) => {
+        return mergeRsbuildConfig(config, {
+          source: {
+            alias: {
+              svelte: path.join(sveltePath, 'src/runtime'),
+            },
+          },
+        });
+      });
+
+      api.modifyBundlerChain(async (chain, { CHAIN_ID, isProd }) => {
```

Comment:
According to the official doc, we may need to add `svelte` to mainFields and conditionNames:

![image](https://github.com/web-infra-dev/rsbuild/assets/7237365/f58e3d4f-bc7e-4e51-8ad4-cdb631fd5413)

https://www.npmjs.com/package/svelte-loader#usage

Reply:
And `resolve.alias`: https://www.npmjs.com/package/svelte-loader#resolvealias

Reply:
> And `resolve.alias`: https://www.npmjs.com/package/svelte-loader#resolvealias

this was configured via `source.alias` with `api.modifyRsbuildConfig`, I have moved it into `api.modifyBundlerChain` to be more consistent.